### PR TITLE
Add a trailing newline to OpenAPI spec output

### PR DESCRIPTION
**Pull request by jgallagher**
Fixes a papercut - if one opens an OpenAPI spec JSON file in an editor that wants to save files with trailing newlines, that breaks later expectorate tests. This changes the output to include a trailing newline instead.

### CODE REVIEW

File: dropshot/src/api_description.rs [hunk 1]
```diff
@@ -1058,9 +1059,11 @@ impl<'a, Context: ServerContext> OpenApiDefinition<'a, Context> {
         out: &mut dyn std::io::Write,
     ) -> serde_json::Result<()> {
         serde_json::to_writer_pretty(
-            out,
+            &mut *out,
             &self.api.gen_openapi(self.info.clone()),
-        )
+        )?;
+        writeln!(out, "").map_err(serde_json::Error::custom)?;
```
Comment (sunshowers) [inline dropshot/src/api_description.rs:1065]:
```suggestion
        writeln!(out).map_err(serde_json::Error::custom)?;
```

**Comment by ahl** [comment]
I don't know if this makes me dumb or a genius, but my editor strips out the trailing newline in json documents. I'm sure I could configure it to preserve them... but it does seem that this may be an annoyance in either direction.

**Comment by jclulow** [comment]
In general I believe most tools expect a UNIX text file to be either totally empty (0 bytes) or a whole number of lines, each terminated by a linefeed.  What text editor is _removing_ them?!

**Comment by ahl** [comment]
Vim

**Comment by sunshowers** [comment]
I think the general expectation is as @jclulow said, to always have a trialing newline yeah. Hence e.g. git shows the "No trailing newline at end of file" message.

**Comment by ahl** [comment]
I see: "an industry standard best practice"!

**Comment by ahl** [comment]
> I got something I'm interpreting as permission-to-merge from Adam offline.

I think my exact words were "over my dead body"

**Pull request merged**